File: LISTAS/LISTA ENLAZADA/LISTAENLAZADA.cs
using System;

public class Nodo {
    public int Dato { get; set; }
    public Nodo Siguiente { get; set; }
    
    public Nodo(int dato) {
        Dato = dato;
        Siguiente = null;
    }
}

public class ListaEnlazada {
    private Nodo cabeza;
    
    public ListaEnlazada() {
        cabeza = null;
    }
    
    public void InsertarInicio(int dato) {
        Nodo nuevo = new Nodo(dato);
        nuevo.Siguiente = cabeza;
        cabeza = nuevo;
    }
    
    public void InsertarFinal(int dato) {
        Nodo nuevo = new Nodo(dato);
        if (cabeza == null) {
            cabeza = nuevo;
            return;
        }
        
        Nodo actual = cabeza;
        while (actual.Siguiente != null) {
            actual = actual.Siguiente;
        }
        actual.Siguiente = nuevo;
    }
    
    public void Eliminar(int dato) {
        if (cabeza == null) return;
        
        if (cabeza.Dato == dato) {
            cabeza = cabeza.Siguiente;
            return;
        }
        
        Nodo actual = cabeza;
        while (actual.Siguiente != null && actual.Siguiente.Dato != dato) {
            actual = actual.Siguiente;
        }
        
        if (actual.Siguiente != null) {
            actual.Siguiente = actual.Siguiente.Siguiente;
        }
    }
    
    public void Mostrar() {
        Nodo actual = cabeza;
        while (actual != null) {
            Console.Write(actual.Dato + " ");
            actual = actual.Siguiente;
        }
        Console.WriteLine();
    }
}

File: LISTAS/LISTA ENLAZADA/LISTAENLAZADA.java
public class LISTAENLAZADA {
    class Nodo {
        int dato;
        Nodo siguiente;
        
        Nodo(int dato) {
            this.dato = dato;
            this.siguiente = null;
        }
    }
    
    private Nodo cabeza;
    
    public LISTAENLAZADA() {
        this.cabeza = null;
    }
    
    public void insertarInicio(int dato) {
        Nodo nuevo = new Nodo(dato);
        nuevo.siguiente = cabeza;
        cabeza = nuevo;
    }
    
    public void insertarFinal(int dato) {
        Nodo nuevo = new Nodo(dato);
        if (cabeza == null) {
            cabeza = nuevo;
            return;
        }
        
        Nodo actual = cabeza;
        while (actual.siguiente != null) {
            actual = actual.siguiente;
        }
        actual.siguiente = nuevo;
    }
    
    public void eliminar(int dato) {
        if (cabeza == null) return;
        
        if (cabeza.dato == dato) {
            cabeza = cabeza.siguiente;
            return;
        }
        
        Nodo actual = cabeza;
        while (actual.siguiente != null && actual.siguiente.dato != dato) {
            actual = actual.siguiente;
        }
        
        if (actual.siguiente != null) {
            actual.siguiente = actual.siguiente.siguiente;
        }
    }
    
    public void mostrar() {
        Nodo actual = cabeza;
        while (actual != null) {
            System.out.print(actual.dato + " ");
            actual = actual.siguiente;
        }
        System.out.println();
    }
}